pkg/controller/plan/migration.go

```diff
@@ -916,7 +943,124 @@ func (r *Migration) resetPrecopyTasks(vm *plan.VMStatus, step *plan.Step) {
 	}
 }
 
-//
+func (r *Migration) createVolumes(vm ref.Ref) (err error) {
+	if r.vmMap == nil {
+		r.vmMap, err = r.kubevirt.VirtualMachineMap()
+		if err != nil {
+			return
+		}
+	}
+	ovirtVm := &ovirt.Workload{}
+	err = r.Source.Inventory.Find(ovirtVm, vm)
+	if err != nil {
+		return
+	}
+	url, err := url.Parse(r.Source.Provider.Spec.URL)
+	if err != nil {
+		return
+	}
+
+	storageName := &r.Context.Map.Storage.Spec.Map[0].Destination.StorageClass
+	for _, da := range ovirtVm.DiskAttachments {
+		populatorCr := v1beta1.OvirtImageIOPopulator{
+			ObjectMeta: meta.ObjectMeta{
+				Name:      da.DiskAttachment.ID,
+				Namespace: r.Plan.Spec.TargetNamespace,
+			},
+			Spec: v1beta1.OvirtImageIOPopulatorSpec{
+				EngineURL:        fmt.Sprintf("https://%s", url.Host),
+				EngineSecretName: r.Source.Secret.Name,
+				DiskID:           da.Disk.ID,
+			},
+		}
+
+		err = r.Client.Create(context.Background(), &populatorCr, &client.CreateOptions{})
+		if err != nil {
+			return
+		}
+		apiGroup := "forklift.konveyor.io"
+
+		pvc := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"kind":       "PersistentVolumeClaim",
+				"apiVersion": "v1",
+				"metadata": map[string]interface{}{
+					"name":      da.DiskAttachment.ID,
+					"namespace": r.Plan.Spec.TargetNamespace,
+				},
+				"spec": map[string]interface{}{
+					"storageClassName": storageName,
+					"resources": map[string]interface{}{
+						"requests": map[string]interface{}{
+							"storage": resource.NewQuantity(da.Disk.ProvisionedSize, resource.BinarySI).String(),
+						},
+					},
+					"accessModes": []string{"ReadWriteOnce"},
+					"dataSourceRef": map[string]interface{}{
+						"apiGroup": apiGroup,
+						"kind":     "OvirtImageIOPopulator",
+						"name":     populatorCr.Name,
+					},
+				},
+			},
+		}
+
+		config, configErr := config.GetConfig()
+		if configErr != nil {
+			return configErr
+		}
+
+		dynamicClient, configErr := dynamic.NewForConfig(config)
+		if configErr != nil {
+			return configErr
+		}
+
+		pvcResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "persistentvolumeclaims"}
+
+		_, err = dynamicClient.Resource(pvcResource).Namespace(r.Plan.Spec.TargetNamespace).Create(context.TODO(), pvc, meta.CreateOptions{})
+
+		if err != nil {
+			return
+		}
+	}
```

Comment:
if we use PVC above, do we need to use dynamic client or the client we already use will be enough?